model/activist.go

```diff
@@ -348,6 +348,106 @@ func HideUser(db *sqlx.DB, userID int) error {
 	return nil
 }
 
+// Merge originalUserID into mergedUserID.
+//  - The original user is hidden
+//  - All of the original user's event attendance is updated to be the merged user.
+func MergeUser(db *sqlx.DB, originalUserID, mergedUserID int) error {
+	if originalUserID == 0 {
+		return errors.New("originalUserID cannot be 0")
+	}
+	if mergedUserID == 0 {
+		return errors.New("mergedUserID cannot be 0")
+	}
+	if originalUserID == mergedUserID {
+		return errors.New("originalUser and mergedUser cannot be the same")
+	}
+
+	tx, err := db.Beginx()
+	if err != nil {
+		return errors.Wrap(err, "could not create transaction")
+	}
+
+	_, err = tx.Exec(`UPDATE activists SET hidden = true WHERE id = ?`, originalUserID)
+	if err != nil {
+		tx.Rollback()
+		return errors.Wrapf(err, "failed to hide original activist %d", originalUserID)
+	}
+
+	// Let's do this the boring way.
+	var originalUserEventIDs []int
+	err = tx.Select(&originalUserEventIDs, `
+SELECT event_id
+FROM event_attendance
+WHERE
+  activist_id = ?
+`, originalUserID)
```

Comment:
You might be able to simplify the logic that follow this query by changing it to exclude any events that the "merged" (target) user attended.

```sql
SELECT event_id
FROM event_attendance ea
WHERE
  activist_id = ? AND
  NOT EXISTS(
	SELECT ea2.event_id
    FROM event_attendance ea2
    WHERE ea2.activist_id = ? 
		AND ea.event_id = ea2.event_id)
```
Where the second query parameter is the "merged" (target) User.

Reply:
done